src/Microsoft.Dnx.Runtime.Sources/Impl/DependencyInjection/ActivatorUtilities.cs

```diff
@@ -76,6 +76,9 @@ public static T CreateInstance<T>(IServiceProvider services)
 
             if (constructors.Length == 1)
             {
+                if (services == null)
+                    throw new InvalidOperationException($"Unable to inject into {type.Name} constructor, current host doesn't support dependency injection");
+
```

Comment:
This is used in MVC for a bunch of unrelated stuff. This fix isn't close enough to what it's claims to address.